mule-extensions-api/src/main/java/org/mule/runtime/extension/api/property/TypeResolversInformationModelProperty.java

```diff
@@ -74,46 +86,55 @@ public String getCategoryName() {
   }
 
   /**
-   * Provides the name of the resolver (if any) associated to a given parameter.
+   * Provides information of the output resolver (if any) associated to the Component.
    *
-   * @param parameterName name of the parameter
-   * @return Name of the resolver associated to the parameter, if one exists.
+   * @return output resolver's {@link ResolverInformation}
    */
-  public Optional<String> getParameterResolverName(String parameterName) {
-    return inputResolvers == null ? empty() : ofNullable(inputResolvers.get(parameterName));
+  public Optional<ResolverInformation> getOutputResolver() {
+    return ofNullable(outputResolver);
   }
 
   /**
-   * Provides the name of the output resolver (if any) associated to the Component.
+   * Provides information the output attributes resolver (if any) associated to the Component.
    *
-   * @return output resolver's name
+   * @return output attributes resolver's {@link ResolverInformation}
    */
-  public Optional<String> getOutputResolverName() {
-    return ofNullable(outputResolver);
+  public Optional<ResolverInformation> getAttributesResolver() {
+    return ofNullable(attributesResolver);
   }
 
   /**
-   * Provides the name of the output attributes resolver (if any) associated to the Component.
+   * Provides information of the metadata keys resolver (if any) associated to the Component.
    *
-   * @return output resolver's name
+   * @return keys resolver's {@link ResolverInformation}
    */
-  public Optional<String> getAttributesResolverName() {
-    return ofNullable(attributesResolver);
+  public Optional<ResolverInformation> getKeysResolver() {
+    return ofNullable(keysResolver);
   }
 
   /**
-   * Provides the name of the metadata keys resolver (if any) associated to the Component.
+   * Provides information of the resolver (if any) associated to a given parameter.
    *
-   * @return output resolver's name
+   * @param parameterName name of the parameter
+   * @return {@link ResolverInformation} of the resolver associated to the parameter, if one exists.
    */
-  public Optional<String> getKeysResolverName() {
-    return ofNullable(keysResolver);
+  public Optional<ResolverInformation> getParameterResolver(String parameterName) {
+    return inputResolvers != null ? ofNullable(inputResolvers.get(parameterName)) : empty();
   }
 
   private String sanitizeResolverName(String resolverName) {
     return isAllBlank(resolverName) || NULL_RESOLVER_NAME.equals(resolverName) ? null : resolverName;
   }
 
+  private ResolverInformation getResolverInformation(String resolverName) {
+    ResolverInformation resolverInformation = null;
+    String sanatizedName = sanitizeResolverName(resolverName);
+    if (sanatizedName != null) {
+      resolverInformation = new ResolverInformation(sanatizedName, true, true);
```

Comment:
why always true, true? if operation is configless this is not true. For Java module this would be false, false.